### PR TITLE
🗳  Show Elections

### PR DESCRIPTION
**Pull request by lprichar**
[//]: # (🚨 Please review the CONTRIBUTING.md in this repository. 💔Thank you!)

### Issue

#87 

### Description
Updates the election list page to show a list of elections, which it pulls from an API call to the back-end.  This borrows heavily from `ElectionTable.tsx`, however that didn't fit the data, so this starts over.

### Testing

1. Ensure the back-end is running
2. Add an election to the back end through postman
3. In the UI click manage elections

You should see something like this:

image

If there are lots of elections it will support pagination like this:

image

It supports filtering:

image

And sorting:

image

### CODE REVIEW

File: packages/admin-app/src/pages/ElectionListPage.tsx
Comment (keithrfung) [inline packages/admin-app/src/pages/ElectionListPage.tsx:17]: This `block` display seems odd. I would just put a Flex in the container and then center everything in that. That will guarantee the full width without all of these pieces to center.

File: packages/admin-app/src/pages/ElectionListPage.tsx
Comment (keithrfung) [inline packages/admin-app/src/pages/ElectionListPage.tsx:51]: Is this a built in?

File: packages/admin-app/src/pages/ElectionListPage.tsx
Comment (keithrfung) [inline packages/admin-app/src/pages/ElectionListPage.tsx:48]: I made a comment about this before but these should all be referencing `MessageId`

Comment (keithrfung) [reply packages/admin-app/src/pages/ElectionListPage.tsx:48]: I would also suggest we make a function so that we refer to intl once in a file and make a helper method so we don't keep adding `id` etc.

File: packages/admin-app/src/pages/ElectionListPage.tsx
Comment (keithrfung) [inline packages/admin-app/src/pages/ElectionListPage.tsx:108]: This should be in Language dictionary.

File: packages/admin-app/src/pages/ElectionListPage.tsx
Comment (keithrfung) [inline packages/admin-app/src/pages/ElectionListPage.tsx:88]: This top level container we should use at the Layout level not the page level.

File: packages/admin-app/src/components/GoHomeButton/GoHomeButton.tsx
Comment (lprichar) [inline packages/admin-app/src/components/GoHomeButton/GoHomeButton.tsx:29]: Even though this go home button is only currently used once, it has so much styling it felt like it should be its own component.  Also, I may grow it into something more reusable as I work on subsequent pages.

**Review by keithrfung — CHANGES_REQUESTED** [review]
Good first go, see my changes

File: packages/api-client/src/server/elections.ts
Comment (lprichar) [inline packages/api-client/src/server/elections.ts:55]: I think this bug is in a lot of the code in this file, but I didn't want to risk breaking anything, so I only changed the code I was using

File: packages/admin-app/src/pages/ElectionListPage.tsx
Comment (lprichar) [inline packages/admin-app/src/pages/ElectionListPage.tsx:44]: You'll notice no manifest or context.  I could be convinced to add those in a separate PR, just let me know.

File: packages/admin-app/src/pages/ElectionListPage.tsx
Comment (lprichar) [reply packages/admin-app/src/pages/ElectionListPage.tsx:88]: Agreed. I'm going to defer implementing this until a subsequent PR to avoid merge conflicts.

Comment (lprichar) [reply packages/admin-app/src/pages/ElectionListPage.tsx]: https://github.com/microsoft/electionguard-ui/issues/107

File: packages/admin-app/src/components/GoHomeButton/index.ts
Comment (keithrfung) [inline packages/admin-app/src/components/GoHomeButton/index.ts:3]: This file doesn't have storybook tests. It technically doesn't need a folder.

File: packages/admin-app/src/routes/RouteIds.ts
Comment (keithrfung) [inline packages/admin-app/src/routes/RouteIds.ts:2]: I know an `Enum` is the first reaction here. Consider a `const` instead. Much more flexible and Enum is getting changed into this object when compiled regardless. 

There is a lot of value for that. Particularly the likelihood of this becoming a function. 

For example:
```js
const routes = {
      home = '/'
      elections = '/elections'
      election_by_id = (id: string) => `/elections/${id}`
}

```

**Review by keithrfung — APPROVED** [review]
A couple nitpicks. Technically doesn't block the PR. Approving, but read over comments.

File: packages/admin-app/src/routes/RouteIds.ts
Comment (lprichar) [reply packages/admin-app/src/routes/RouteIds.ts]: Good call, that makes a lot of sense, thanks

**Pull request merged**
